apis/v1/httproute_types.go

```diff
@@ -305,7 +305,7 @@ type HTTPRouteTimeouts struct {
 	//
 	// For example, setting the `rules.timeouts.request` field to the value `10s` in an
 	// `HTTPRoute` will cause a timeout if a client request is taking longer than 10 seconds
-	// to complete.
+	// to complete. Specifying a zero value such as "0s" is interpreted as no timeout.
```

Comment:
This is probably the simplest way to say this, but just to be very precise here, I don't think many/any implementations can really support an infinite timeout. Maybe it would be more precise to say that this should be set to the maximum value an implementation can support?

/cc @frankbu @kflynn

Reply:
This language is already in the spec - I just copied it from the top the godoc for `HTTPRouteTimeouts`

Reply:
I'm actually kind of accustomed to many timeouts being able to be completely disabled, but @robscott's point is well made: we probably do want to allow an implementation that can't completely disable the timeout to set it to Something Very Long™. Maybe

> Setting a timeout to the zero duration (e.g. "0s") SHOULD disable the timeout completely. Implementations that cannot completely disable the timeout MUST instead interpret the zero duration as the longest possible value to which the timeout can be set.

(Note that this is currently duplicated on lines 298 and 308; my vote would be to remove it on 298 if it's included on 308, but if not, we should change it in both places.)

Reply:
FYI - line 298 doesn't show up on the website/crd because the HTTPRouteRule.Timeouts has it's own godoc where the struct is used as a field.

I think I'd prefer to annotate each Timeout option separately so leaving the comment on 308 & 326

Reply:
+1 to @kflynn's suggestion here + @dprotaso's suggestion to include the text ~everywhere.

Reply:
updated